src/MissionManager/PlanManager.h

```diff
@@ -68,9 +68,9 @@ class PlanManager : public QObject
 
     // These values are public so the unit test can set appropriate signal wait times
     // When passively waiting for a mission process, use a longer timeout.
-    static const int _ackTimeoutMilliseconds = 1500;
+    static const int _ackTimeoutMilliseconds = 2500;
```

Comment:
This seems super long. This would only take affect on things like a go to location command. Are you sure this is needed? For mission download only the retry timeout comes into play.

Reply:
I had to build in debug goal to get logging again. This is the section when uploading mission to vehicle without touching the regular timeout.
```
PlanMasterControllerLog: PlanMasterController::sendToVehicle start mission sendToVehicle
MissionControllerLog: MissionControllerLog::sendToVehicle
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 0 0 "Mission Start"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 1 1 "Takeoff"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 2 2 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 3 3 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 4 4 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 5 5 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 6 8 "Landing Pattern"
PlanManagerLog: "writeMissionItems T:Mission count:" 9
PlanManagerLog: _setTransactionInProgress "T:Mission" 2
PlanManagerLog: "_writeMissionCount T:Mission count:_retryCount" 9 0
PlanManagerLog: "Retrying T:Mission MISSION_COUNT retry Count" 1
PlanManagerLog: "_writeMissionCount T:Mission count:_retryCount" 9 1
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 0 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequence number requested which has already been sent, sending again:" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 0 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 1
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 1 22
QString::arg: Argument missing: _handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE, T:Mission
PlanManagerLog: "_handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 2
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 2 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 2
PlanManagerLog: "_handleMissionRequest T:Mission sequence number requested which has already been sent, sending again:" 2
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 2 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 3
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 3 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 3
PlanManagerLog: "_handleMissionRequest T:Mission sequence number requested which has already been sent, sending again:" 3
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 3 16
QString::arg: Argument missing: _handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE, T:Mission
PlanManagerLog: "_handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 4
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 4 16
QString::arg: Argument missing: _handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE, T:Mission
PlanManagerLog: "_handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 4
PlanManagerLog: "_handleMissionRequest T:Mission sequence number requested which has already been sent, sending again:" 4
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 4 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 5
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 5 16
QString::arg: Argument missing: _handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE, T:Mission
PlanManagerLog: "_handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 5
PlanManagerLog: "_handleMissionRequest T:Mission sequence number requested which has already been sent, sending again:" 5
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 5 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 6
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 6 189
QString::arg: Argument missing: _handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE, T:Mission
PlanManagerLog: "_handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 7
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 7 31
PlanManagerLog: "Sending error - _planTypeString(T:Mission) errorCode(2) errorMsg(Vehicle did not request all items from ground station: MISSION_REQUEST)"
PlanManagerLog: _setTransactionInProgress "T:Mission" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle start GeoFence sendToVehicle
PlanManagerLog: "writeMissionItems T:GeoFence count:" 0
PlanManagerLog: _setTransactionInProgress "T:GeoFence" 2
PlanManagerLog: "_writeMissionCount T:GeoFence count:_retryCount" 0 0
PlanManagerLog: "_handleMissionRequest T:GeoFence Incorrect mission_type received expected:actual" 1 0
PlanManagerLog: "_handleMissionRequest T:GeoFence Incorrect mission_type received expected:actual" 1 0
PlanManagerLog: "_handleMissionAck T:GeoFence type:" "Mission accepted."
PlanManagerLog: "_handleMissionAck write sequence complete T:GeoFence"
PlanManagerLog: _setTransactionInProgress "T:GeoFence" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle start rally sendToVehicle
PlanManagerLog: "writeMissionItems T:Rally count:" 0
PlanManagerLog: _setTransactionInProgress "T:Rally" 2
PlanManagerLog: "_writeMissionCount T:Rally count:_retryCount" 0 0
PlanManagerLog: "_handleMissionAck T:Rally type:" "Mission accepted."
PlanManagerLog: "_handleMissionAck write sequence complete T:Rally"
PlanManagerLog: _setTransactionInProgress "T:Rally" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle start mission sendToVehicle
MissionControllerLog: MissionControllerLog::sendToVehicle
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 0 0 "Mission Start"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 1 1 "Takeoff"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 2 2 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 3 3 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 4 4 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 5 5 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 6 8 "Landing Pattern"
PlanManagerLog: "writeMissionItems T:Mission count:" 9
PlanManagerLog: _setTransactionInProgress "T:Mission" 2
PlanManagerLog: "_writeMissionCount T:Mission count:_retryCount" 9 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 8
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 8 21
PlanManagerLog: "Fell off end of switch statement T:Mission 15"
PlanManagerLog: "_handleMissionAck T:Mission type:" "Unknown error: 15. Item #8 Command: Land"
PlanManagerLog: "Fell off end of switch statement T:Mission 15"
PlanManagerLog: "Sending error - _planTypeString(T:Mission) errorCode(5) errorMsg(Unknown error: 15. Item #8 Command: Land)"
PlanManagerLog: _setTransactionInProgress "T:Mission" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle start GeoFence sendToVehicle
PlanManagerLog: "writeMissionItems T:GeoFence count:" 0
PlanManagerLog: _setTransactionInProgress "T:GeoFence" 2
PlanManagerLog: "_writeMissionCount T:GeoFence count:_retryCount" 0 0
PlanManagerLog: "Sending error - _planTypeString(T:GeoFence) errorCode(2) errorMsg(Mission write failed, vehicle failed to send final ack.)"
PlanManagerLog: _setTransactionInProgress "T:GeoFence" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle start rally sendToVehicle
PlanManagerLog: "writeMissionItems T:Rally count:" 0
PlanManagerLog: _setTransactionInProgress "T:Rally" 2
PlanManagerLog: "_writeMissionCount T:Rally count:_retryCount" 0 0
PlanManagerLog: "_handleMissionRequest T:Rally Incorrect mission_type received expected:actual" 2 0
PlanManagerLog: "_handleMissionAck T:Rally Incorrect mission_type received expected:actual" 2 0
PlanManagerLog: "_handleMissionAck T:Rally Incorrect mission_type received expected:actual" 2 1
PlanManagerLog: "_handleMissionAck T:Rally type:" "Mission accepted."
PlanManagerLog: "_handleMissionAck write sequence complete T:Rally"
PlanManagerLog: _setTransactionInProgress "T:Rally" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle start mission sendToVehicle
MissionControllerLog: MissionControllerLog::sendToVehicle
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 0 0 "Mission Start"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 1 1 "Takeoff"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 2 2 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 3 3 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 4 4 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 5 5 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 6 8 "Landing Pattern"
PlanManagerLog: "writeMissionItems T:Mission count:" 9
PlanManagerLog: _setTransactionInProgress "T:Mission" 2
PlanManagerLog: "_writeMissionCount T:Mission count:_retryCount" 9 0
PlanManagerLog: "Retrying T:Mission MISSION_COUNT retry Count" 1
PlanManagerLog: "_writeMissionCount T:Mission count:_retryCount" 9 1
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 0 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequence number requested which has already been sent, sending again:" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 0 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 1
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 1 22
PlanManagerLog: "Sending error - _planTypeString(T:Mission) errorCode(2) errorMsg(Vehicle did not request all items from ground station: MISSION_REQUEST)"
PlanManagerLog: _setTransactionInProgress "T:Mission" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle start GeoFence sendToVehicle
PlanManagerLog: "writeMissionItems T:GeoFence count:" 0
PlanManagerLog: _setTransactionInProgress "T:GeoFence" 2
PlanManagerLog: "_writeMissionCount T:GeoFence count:_retryCount" 0 0
PlanManagerLog: "_handleMissionAck T:GeoFence Incorrect mission_type received expected:actual" 1 0
PlanManagerLog: "_handleMissionRequest T:GeoFence Incorrect mission_type received expected:actual" 1 0
PlanManagerLog: "_handleMissionAck T:GeoFence type:" "Mission accepted."
PlanManagerLog: "_handleMissionAck write sequence complete T:GeoFence"
PlanManagerLog: _setTransactionInProgress "T:GeoFence" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle start rally sendToVehicle
PlanManagerLog: "writeMissionItems T:Rally count:" 0
PlanManagerLog: _setTransactionInProgress "T:Rally" 2
PlanManagerLog: "_writeMissionCount T:Rally count:_retryCount" 0 0
PlanManagerLog: "_handleMissionRequest T:Rally Incorrect mission_type received expected:actual" 2 0
PlanManagerLog: "_handleMissionAck T:Rally type:" "Mission accepted."
PlanManagerLog: "_handleMissionAck write sequence complete T:Rally"
PlanManagerLog: _setTransactionInProgress "T:Rally" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle start mission sendToVehicle
MissionControllerLog: MissionControllerLog::sendToVehicle
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 0 0 "Mission Start"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 1 1 "Takeoff"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 2 2 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 3 3 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 4 4 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 5 5 "Waypoint"
MissionControllerLog: _convertToMissionItems seqNum:lastSeqNum:command 6 8 "Landing Pattern"
PlanManagerLog: "writeMissionItems T:Mission count:" 9
PlanManagerLog: _setTransactionInProgress "T:Mission" 2
PlanManagerLog: "_writeMissionCount T:Mission count:_retryCount" 9 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 2
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 2 16
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber" 0
PlanManagerLog: "_handleMissionRequest T:Mission sequenceNumber:command" 0 16
QString::arg: Argument missing: _handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE, T:Mission
PlanManagerLog: "_handleMissionAck ArduPilot sending possibly bogus MAV_MISSION_INVALID_SEQUENCE" 0
PlanManagerLog: "Sending error - _planTypeString(T:Mission) errorCode(2) errorMsg(Vehicle did not request all items from ground station: MISSION_REQUEST)"
PlanManagerLog: _setTransactionInProgress "T:Mission" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle start GeoFence sendToVehicle
PlanManagerLog: "writeMissionItems T:GeoFence count:" 0
PlanManagerLog: _setTransactionInProgress "T:GeoFence" 2
PlanManagerLog: "_writeMissionCount T:GeoFence count:_retryCount" 0 0
PlanManagerLog: "_handleMissionRequest T:GeoFence Incorrect mission_type received expected:actual" 1 0
PlanManagerLog: "_handleMissionAck T:GeoFence type:" "Mission accepted."
PlanManagerLog: "_handleMissionAck write sequence complete T:GeoFence"
PlanManagerLog: _setTransactionInProgress "T:GeoFence" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle start rally sendToVehicle
PlanManagerLog: "writeMissionItems T:Rally count:" 0
PlanManagerLog: _setTransactionInProgress "T:Rally" 2
PlanManagerLog: "_writeMissionCount T:Rally count:_retryCount" 0 0
PlanManagerLog: "_handleMissionRequest T:Rally Incorrect mission_type received expected:actual" 2 0
PlanManagerLog: "_handleMissionAck T:Rally type:" "Mission accepted."
PlanManagerLog: "_handleMissionAck write sequence complete T:Rally"
PlanManagerLog: _setTransactionInProgress "T:Rally" 0
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete
PlanMasterControllerLog: PlanMasterController::sendToVehicle Rally Point send complete

```

Reply:
I have to amend my initial comment. I just had the same issue when just uploading a mission that already is on the vehicle but I have a wireshark recording from the issue.